src/volume.ts

```diff
@@ -238,7 +264,7 @@ export class Volume extends Construct implements IStorage {
   };
 
 
-  private constructor(scope: Construct, id: string,
+  protected constructor(scope: Construct, id: string,
```

Comment:
How come?

Reply:
It's either that, or a static constructor that accepts `k8s.Volume` (which is basically making this constructor public.

One of the more frustrating things I found about this is that the class is completely locked down to extension while not offering all option that k8s API currently has.

Reply:
Dropped in favor of changing IStorage